test/e2e/scheduling/preemption.go

```diff
@@ -341,11 +343,13 @@ var _ = SIGDescribe("PodPriorityResolution [Serial] [Feature:PodPreemption]", fu
 				Namespace:         metav1.NamespaceSystem,
 				PriorityClassName: spc,
 			})
+			defer func() {
```

Comment:
The previous defer kind of made sense, b/c there were error conditions that could be hit before the cleanup.  This one makes less sense to me.  Is there a specific condition you are looking to prevent?

Reply:
Since there is an `Expect` statement before the clean up I considered there was a possibility for the test to finish without calling deleting the pod.

Reply:
This is inside a loop. This is not equivalent to the original code. These pods would no longer be deleted inside the loop. Their deletion would be deferred to the time that the function returns.

Reply:
> These pods would no longer be deleted inside the loop. 

There'll be just one additional BE pod here, so I think that should be fine :)